src/Microsoft.Azure.ServiceBus/MessageReceiver.cs

```diff
@@ -111,9 +131,20 @@ public Task<DateTime> RenewLockAsync(Guid lockToken)
             return this.OnRenewLockAsync(lockToken);
         }
 
-        internal Task<AmqpResponseMessage> ExecuteRequestResponseAsync(AmqpRequestMessage amqpRequestMessage)
+        public Task<BrokeredMessage> PeekAsync()
```

Comment:
@jtaubensee , I agree we need to have documentation for all the public methods (new and old). I will send a PR later with the documentation. Its better to get the whole thing written together for uniformity.